spec/board_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require './lib/tic_tac_toe.rb'
+
+RSpec.describe Board do
+  describe ' #initialize_Board ' do
+    it ' test the correct number on each cell of the grid' do
+      board = Board.new
+      cond = true
+      m_board = board.main_board.flatten
+      m_board.each_with_index do |cell, index|
+        cond = cell.to_i == (index + 1)
+        break unless cond
+      end
+      expect(cond).to be true
+    end
+  end
+
+  describe '#check space' do
+    it 'returns true when a space is availabe in the board' do
+      board = Board.new
+      cond = board.space?
+      expect(cond).to be true
+    end
+
+    it 'returns false when a space is not availabe in the board' do
+      board = Board.new
+      9.times do |n|
+        board.save_choise_player('X', n + 1)
+      end
+      cond = board.space?
+      expect(cond).to be false
+    end
+  end
+
+  describe ' #check_choise_player' do
+    it ' when a player select a number of the empty grid ' do
+      board = Board.new
```

Comment:
#### `REQUIRED`

Following the above suggestion, this line is not needed anymore.